### PR TITLE
Replace `AccountsBalanceTracker` with accounting deltas

### PR DESCRIPTION
**Pull request by azarovh**
`AccountsBalanceTracker` was used to check if multiple account spend in a single tx won't underflow the balance. To avoid duplicating code when adding support for Orders I use Deltas now. They already implement all the constraint check and are easily reusable for the job.

### CODE REVIEW

File: chainstate/constraints-value-accumulator/src/constraints_accumulator.rs [hunk 1]
```diff
@@ -72,7 +70,11 @@ impl ConstrainedValueAccumulator {
 
         let mut accumulator = Self::new();
         let mut total_to_deduct = BTreeMap::<CoinOrTokenId, Amount>::new();
-        let mut accounts_balances_tracker = AccountsBalancesTracker::new(pos_accounting_view);
+
+        // Temp deltas are used to check accounting errors like overspends across multiple inputs
+        let mut temp_pos_accounting = pos_accounting::PoSAccountingDelta::new(pos_accounting_view);
+        let mut temp_orders_accounting =
+            orders_accounting::OrdersAccountingCache::new(orders_accounting_view);
```
Comment (TheQuantumPhysicist) [inline chainstate/constraints-value-accumulator/src/constraints_accumulator.rs:78]: What about tokens accounting?

Comment (azarovh) [reply chainstate/constraints-value-accumulator/src/constraints_accumulator.rs]: done

File: api-server/scanner-lib/src/sync/tests/simulation.rs [hunk 2]
```diff
@@ -113,6 +121,34 @@ impl<'a> PoSAccountingView for PoSAccountingAdapterToCheckFees<'a> {
     }
 }
 
+struct StubTokensAccounting;
```
Comment (TheQuantumPhysicist) [inline api-server/scanner-lib/src/sync/tests/simulation.rs:124]: I thought we used the word "Mock" instead of "Stub"... it's a matter of convention.

**Review by TheQuantumPhysicist — APPROVED** [review]

File: api-server/scanner-lib/src/sync/tests/simulation.rs (hunk 2 above)
Comment (azarovh) [reply api-server/scanner-lib/src/sync/tests/simulation.rs:124]: Idk it's actually a stub with hardcoded state to satisfy function signature. It doesn't have any logic or expectation as mocks

**Pull request merged**